[new] Add Kule Lazy to Lib

keicheng opened the pull request.

PeterDaveHello commented:
As the first commit to add a lib, would you please also add the laster version with its files? Thanks.

keicheng commented:
I used cdnjs-importer and finished commands, but i don`t know how to push

PeterDaveHello commented:
If you are not so familiar with this project, we can do it for you, but I'll need to add tags in its repo for each version, thanks.

keicheng commented:
Thanks, let me try again

PeterDaveHello commented:
Looks better now, will there be a stable release soon?

keicheng commented:
Yes, it will.
I dont`t know which step is miss or wrong /_\

PeterDaveHello commented:
@keicheng I will take a look at it later, but as far as i know circleci has problem with their new structure for a few days and the problem has not been fixed yet.

keicheng commented:
ok, thanks. i`ll try again later.

PeterDaveHello commented:
@keicheng You can also run `npm test` locally.

keicheng commented:
I tried, it echo "Error: no test specified",
I google it, but not answer.

PeterDaveHello commented:
@keicheng Should do it under cdnjs folder.

keicheng commented:
I done on cdnjs folder.
/cdnjs? or /cdnjs/ajax/libs/kule.lazy?
Both path all wrong /_\

I need to edit some files?

PeterDaveHello commented:
@keicheng root of cdnjs, have you tried `npm install` to install its dependencies first?

keicheng commented:
yes, i installed vows and vows-si, but it echo Error: Cannot find module 'vows-si'

keicheng commented:
npm test is ok after npm update, thanks.

PeterDaveHello commented:
@keicheng nice, i will help you review your pull request later, before that, you should squash your commits into only one commit via `git rebase` and it will be even better if you can sync/rebase your branch with our latest master branch, after that, push your commit with `--force` parameter to update this pull request, then everything will be very good!

PeterDaveHello commented:
@keicheng again, if you don't wanna do these works, i can still add it for you by my side, or you can also learn some git and workflow skills by make it pretty! It is all up to you. No worries.

keicheng commented:
i nerver edit "ajax/libs/Sortable/package.json",
but,  after git status, echo "modified:   ajax/libs/Sortable/package.json"

PeterDaveHello commented:
@keicheng I guess you are using Windows or Mac OS X right now, am I correct? Sparse checkout may help! See: https://github.com/cdnjs/cdnjs/blob/master/sparseCheckout.md

keicheng commented:
Yes, i use MAC OS X, i try it.

keicheng commented:
I try npm test that is ok, but circleci still has failed /_\

keicheng commented:
@PeterDaveHello 很感謝你細心又有耐心的指導我，我天資駑鈍最後還是 Failed 了

PeterDaveHello commented:
@keicheng As i said, should be circleci's preblem, no worries.

PeterDaveHello commented on ajax/libs/kule.lazy/package.json:
```diff
@@ -0,0 +1,35 @@
+    "mobile",
+    "tiles"
+  ],
+  "filename": "CHANGELOG.md",
```
Filename should point to the main file.

PeterDaveHello commented on ajax/libs/kule.lazy/package.json:
```diff
@@ -0,0 +1,35 @@
+    "url": "git://github.com/keicheng/kule.lazy.git"
+  },
+  "author": "Kei Cheng",
+  "license": "Free",
```
Do you have any license document about "Free" license?

keicheng replied:
Not yet. But like MIT, i don`t know I need to submit somthing?

PeterDaveHello replied:
If it's not confirmed, I think we can remove this line.

keicheng replied:
I change to MIT and edited it.
http://www.kule.tw/intro.html#license

PeterDaveHello replied:
Great!

keicheng commented:
Passed? really? oh~yah~

PeterDaveHello commented:
@keicheng but I'll need you use `git merge` to squash them into only one commit, and rebase with our latest master branch, thanks!

keicheng commented:
i try it

keicheng commented:
Sorry, I don`t know how rebase your master branch, and squash commit

PeterDaveHello commented:
@keicheng `git rebase -i HEAD~20` will work in this case!

PeterDaveHello commented:
@keicheng If you want me do that for you, you can ping me any time.

keicheng commented:
after "git rebase -i HEAD~20", i edit commit content? right?

PeterDaveHello commented:
@keicheng To squash them, you should find all the commits "after" your first commit, then repalce their "pick" to "squash", then git will help you squash them into only one commit!

PeterDaveHello commented:
@keicheng To reabse with our latest master branch, see this: https://www.peterdavehello.org/2015/05/how-to-syncupdate-forked-repository-with-upstream/

keicheng commented:
@PeterDaveHello  Sorry, 雖然重新提交了，但我還是想試著想把 squash 弄清楚，不曉得這樣是否正確

PeterDaveHello commented:
@keicheng please squash them into only "one commit"

keicheng commented:
還是不太曉得 squash 怎麼操作

keicheng commented:
@PeterDaveHello Hi, 不曉得是不是這樣？

PeterDaveHello commented:
Should update commit message.

keicheng commented:
ok, thanks

keicheng commented:
0rz...

Amomo commented on ajax/libs/kule.lazy/package.json:
```diff
@@ -0,0 +1,39 @@
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/keicheng/kule.lazy.git",
+    "basePath": "",
```
I think the "basePath" should point to `"css"` folder.
That your filemap would match to your commit.

PeterDaveHello replied:
or change `files` to `css/**/*`

keicheng replied:
but, i no have css folder
can i set that?

i try it

Amomo replied:
Here is a git autoupdate config.
cdnjs would update the library when its GitHub repo add a new version tag.

And, I found that your library on GitHub don't have tags yet.
Could you add the tag in your GitHub repo? Thanks!

Amomo commented:
Hi @keicheng 

Recently, the build sometimes get failed before the test starting.
Could you try to resend your commit before the problem fixed?
Sorry for the inconvenience.

keicheng commented:
but, i no have css folder

keicheng commented:
oh~ That`s alright?

PeterDaveHello commented:
Should modify the commit message to fit the work.

keicheng commented:
commit message is ok?

PeterDaveHello commented:
Commit message okay, since you change `files` to `"css/**/*"`, you need to put the current files under a `css` folder.

Amomo commented:
Hi @keicheng,  "basePath" is like a starting point in your git repo.

If you use `"basePath": ""`, `"files": "css/**/*"`
Your files in commit should under **ajax/libs/kule.lazy/3.0.1007beta/`css/`**
That the filemap in `"files"` would be same with your GitHub repo.

keicheng commented:
i done, but i`m not sure it`s correct or not

PeterDaveHello commented:
Looks okay, @cdnjs/intern please double check for this PR, and the last thing will be rebase this branch with our latest master branch, thanks!

maruilian11 commented:
i think the PR is ok~
only need to rebase :+1:

keicheng commented:
i try to rebase from upstream/master, but have some conflicts. i don`t know how to fixed, maybe i have wrong steps?

PeterDaveHello commented:
Don't know, you can show your error message.

keicheng commented:
[kei@KeiMBP ~/Works/github/cdnjs]#git rebase upstream/master
First, rewinding head to replay your work on top of it...

Applying: Cut the ribbon, welcome to cdnjs.com
Applying: Example package.json in progress
Applying: Had an error in directory structure, lib -> libs
Applying: backbone.js renamed to package.json as a standard library descriptor
Using index info to reconstruct a base tree...
Falling back to patching base and 3-way merge...
Removing ajax/libs/backbone.js/backbone.js
Applying: Added backbone.js version 0.3.3 and implemented first attempt at package.json
Using index info to reconstruct a base tree...
M	ajax/libs/backbone.js/package.json
<stdin>:102: trailing whitespace.
       "events"
<stdin>:108: trailing whitespace.
           "web": "http://ashkenas.com/"
<stdin>:109: trailing whitespace.
       }
<stdin>:115: trailing whitespace.
           "web": "http://www.embedthis.com"
<stdin>:116: trailing whitespace.
       }
warning: squelched 2 whitespace errors
warning: 7 lines add whitespace errors.
Falling back to patching base and 3-way merge...
warning: Cannot merge binary files: ajax/libs/backbone.js/0.3.3/backbone-min.js (cf5e96788e0db25d3a1efb5a91dba69b098c0850 vs. Added backbone.js version 0.3.3 and implemented first attempt at package.json)
Auto-merging ajax/libs/backbone.js/package.json
CONFLICT (content): Merge conflict in ajax/libs/backbone.js/package.json
Auto-merging ajax/libs/backbone.js/0.3.3/backbone-min.js
CONFLICT (add/add): Merge conflict in ajax/libs/backbone.js/0.3.3/backbone-min.js
Failed to merge in the changes.
Patch failed at 0005 Added backbone.js version 0.3.3 and implemented first attempt at package.json
The copy of the patch that failed is found in:
   /Users/kei/Works/github/cdnjs/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

keicheng commented:
if i re-fork and redo all steps, it will easy?
i think my fork repo is too old.

PeterDaveHello commented:
@keicheng Yes it may, but you look like wanna learn some git skills, maybe we can help you fix it.

keicheng commented:
Thanks for let me learn git skills, it will help me more in future.
I already deleted old fork repo, but I`ll to find and study method for this problem.
Thank you very very very much.

PeterDaveHello commented:
@keicheng no problem, I'm glad you didn't hate it, @cdnjs/intern please audit this pR

keicheng commented:
Git is great tool and cdnjs is good service, I like both.
You are good guys and you from Taiwan, it`s important! hahaha

I already done, hope it`s final
https://github.com/cdnjs/cdnjs/pull/5917

PeterDaveHello commented:
Ok.
